lib/pdf_fill/forms/va210781.rb

```diff
@@ -105,22 +300,126 @@ def merge_fields
 
       private
 
-      def expand_veteran_full_name
-        @form_data['veteranFullName'] = extract_middle_i(@form_data, 'veteranFullName')
+      def expand_incidents(incidents)
+        return if incidents.blank?
+        incidents.each_with_index do |incident, index|
+          format_incident_overflow(incident, index + 1)
+          incident['incidentDate'] = expand_incident_date(incident)
+          expand_unit_assigned_dates(incident)
+          incident['incidentLocation'] = expand_incident_location(incident)
+          incident['unitAssigned'] = expand_incident_unit_assignment(incident)
+          expand_persons_involved(incident)
+        end
       end
 
-      def expand_ssn
-        ssn = @form_data['veteranSocialSecurityNumber']
-        return if ssn.blank?
-        ['', '1', '2'].each do |suffix|
-          @form_data["veteranSocialSecurityNumber#{suffix}"] = split_ssn(ssn)
+      def expand_persons_involved(incident)
+        return if incident.blank?
+        return if incident['personInvolved'].blank?
+
+        persons_involved = incident['personInvolved']
+        persons_involved.each_with_index do |person_involved, index|
+          expand_injury_death_date(person_involved, index)
+          split_person_unit_assignment(person_involved, index)
+          flatten_person_identification(person_involved, index)
+          resolve_cause_injury_death(person_involved, index)
+        end
+
+        combined_persons_involved = {}
+        persons_involved.each do |person|
+          person.each do |key, value|
+            combined_persons_involved[key] = value
+          end
```

Comment:
If I'm reading this correctly, `persons_involved` is structured as an array of hashes, like `[{"foo"=>1}, {"bar"=>2}]`. If that's correct, this can be shortened to `Hash[*persons_involved.collect{|h| h.to_a}.flatten]`